common/table.js

```diff
@@ -806,6 +806,36 @@
                 _callonSelectedRowsChanged(scope, [tuple], isSelected);
             };
 
+            // NOTE: For some reason this is being registered/triggered twice when the download widget is part of the record table (main body of record)
+            $timeout(function () {
+                var downloadElements = angular.element(".download");
+                downloadElements.on('click', function (e) {
+                    e.preventDefault();
+
+                    if (!e.target.inProgress) {
+                        e.target.inProgress = true;
+
+                        var spinnerHTML = ' <span class="glyphicon glyphicon-refresh glyphicon-refresh-animate"></span>';
+                        //show spinner
+                        e.target.innerHTML += spinnerHTML;
+
+                        var dcctx = ConfigUtils.getContextJSON();
+                        // make a HEAD request to check if the user can fetch the file
+                        dcctx.server.http.head(e.target.href).then(function (response) {
+                            // fetch the file for the user
+                            $window.open(e.target.href, '_blank');
+                        }).catch(function (exception) {
+                            // If an error occurs while a user is trying to download the file, allow them to dismiss the dialog
+                            ErrorService.handleException(exception, true);
```

Comment:
You should call `ERMrest.responseToError` here.

Reply:
Also please make sure to test the 404 case too. I'm not sure whether the error message that we're showing makes sense in this case or not.